server/datastore/mysql/migrations/tables/20220915153947_HostDisplayName.go

```diff
@@ -0,0 +1,40 @@
+package tables
+
+import (
+	"database/sql"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20220915153947, Down_20220915153947)
+}
+
+func Up_20220915153947(tx *sql.Tx) error {
+	for _, change := range []struct{ name, sql string }{
+		{"delete index", `ALTER TABLE hosts DROP INDEX hosts_search`},
+		{"create index", `CREATE FULLTEXT INDEX hosts_search ON hosts(hostname, uuid, computer_name)`},
+		{"new table", `
+			CREATE TABLE host_display_name (
+			    host_id int(10) unsigned NOT NULL,
+			    display_name varchar(255) NOT NULL,
+			    PRIMARY KEY (host_id),
+			    KEY (display_name)
+			);
+		`},
+		{"migrate data", `
+			INSERT INTO host_display_name (
```

Comment:
This might be a costly migration for some users, I know we've been very careful in other cases with migrating data from `hosts` to secondary tables, using a lazy approach. I'm not sure if it's possible in that case (the display name might only be set at host enroll time and never be updated again?), so I guess the only thing would be to load-test the migration with e.g. 100K hosts and see how badly it impacts the downtime.